### DISCUSSION TITLE
Question about dimension of y after applying mask in PixArt.py

### DISCUSSION THREAD

**Discussion by Starry-lei** [PixArt-alpha/PixArt-alpha#136] [Q&A]
In the code: "if mask is not None:
            if mask.shape[0] != y.shape[0]:
                mask = mask.repeat(y.shape[0] // mask.shape[0], 1)
            mask = mask.squeeze(1).squeeze(1)
            y = y.squeeze(1).masked_select(mask.unsqueeze(-1) != 0).view(1, -1, x.shape[-1])
            y_lens = mask.sum(dim=1).tolist()"
            
            why is  y viewed as (1, -1, x.shape[-1]) instead of (N, -1, x.shape[-1]) ?

**Comment by lawrence-cj** [comment]
later used in xformers
